app/services/price_policies/time_based_price_calculator.rb

```diff
@@ -16,8 +16,14 @@ def initialize(price_policy)
     def calculate(start_at, end_at)
       return if start_at > end_at
       duration_mins = TimeRange.new(start_at, end_at).duration_mins
-      discount_multiplier = calculate_discount(start_at, end_at)
-      cost_and_subsidy(duration_mins, discount_multiplier)
+
+      if product.is_a?(Instrument) && product.duration_pricing_mode? && product.duration_rates.present?
+        # Should return something for subsidy as well?
```

Comment:
My guess is we'll need this to also return the `subsidy` and the calculation will either be `duration_in_hours * usage_subsidy` or just `0` if it doesn't make sense for this case

Reply:
I think that the return value needs to include subsidy, but I'm not clear if it should be 0 or something else